src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -72,6 +72,7 @@ private void StopRaisingEvents()
             {
                 _cancellation = null;
                 token.Cancel();
+                token.Dispose();
```

Comment:
@wfurt I have a question: why do we *only* cancel and dispose the `token` here? The RunningInstance class receives a copy of this token, so why not perform these two actions inside `CancellationCallback` too? (follow up of Marius' question).

Reply:
CancellationCallback() did not do anything with the token. It is StartRaisingEvents() where token was created and symmetrically StopRaisingEvents() where token is destroyed. Existing `_cancellation = null` should be sufficient IMHO to dispose token. I only add explicit Dispose() call to make it explicit and not wait for GC. I did not want to introduce different life cycle for the token and worry about what changed.